app/models/user_group.rb

```diff
@@ -53,7 +53,9 @@ def self.text_search(text)
   def add(users)
     # Force individual queries for paper_trail
     Array.wrap(users).uniq.each do |user|
-      user_group_members.where(user_id: user.id).first_or_create!
+      member = user_group_members.with_deleted.where(user_id: user.id).first_or_create!
+      member.restore if member.deleted?
```

Comment:
@gigxz do you think we should have some tests for this scenario where there's a deleted record being restored?

Reply:
I added some assertions to https://github.com/greenriver/hmis-warehouse/pull/4474/commits/6c55ed4cfe002db73aacf9c15e5a3100e47d4804

Reply:
Thank you!